### PR TITLE
fix: support receive special function

### PR DESCRIPTION
**Pull request by magicalne**
[The `receive` function](https://docs.soliditylang.org/en/v0.8.13/contracts.html?highlight=receive#receive-ether-function) is executed on a call to the contract with empty calldata. 

Polyjuice didn't allow zero-size input. So people cannot use the `receive ether function`. 
This PR fix the issue and some test cases need to be fixed due to compatibility issues.

### CODE REVIEW

File: c/polyjuice.h [hunk 1]
```diff
@@ -1150,7 +1150,7 @@ int handle_message(gw_context_t* ctx,
 
   debug_print_int("[handle_message] msg.kind", msg.kind);
   /* NOTE: msg and res are updated */
-  if (to_address_exists && code_size > 0 && (is_create(msg.kind) || msg.input_size > 0)) {
+  if (to_address_exists && code_size > 0 && (is_create(msg.kind) || msg.input_size >= 0)) {
```
Comment (TheWaWaR) [inline c/polyjuice.h:1153]: `msg.input_size` has type `size_t` which is an unsigned integer type and can not be less then `0`. So just remove the `|| msg.input_size >= 0` check.

**Comment by Flouse** [comment]
Please fix the [failure tests](https://github.com/nervosnetwork/godwoken-polyjuice/runs/6663659053?check_suite_focus=true#step:8:1370):
    test_cases::address_collision::create2_address_collision_duplicate
    test_cases::address_collision::create2_address_collision_overwrite

File: polyjuice-tests/src/test_cases/receive_ether.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+use crate::helper::{
+    check_cycles, create_block_producer, deploy, new_block_info, parse_log, setup,
+    MockContractInfo, PolyjuiceArgsBuilder, CREATOR_ACCOUNT_ID, L2TX_MAX_CYCLES,
+};
+
+const INIT_CODE: &str = include_str!("./evm-contracts/EtherReceiverMock.bin");
+
+#[test]
+fn receive_ether_test() -> anyhow::Result<()> {
```
Comment (Flouse) [inline polyjuice-tests/src/test_cases/receive_ether.rs:18]: Please add a test case:
If neither a receive Ether nor a payable fallback function is present, the contract cannot receive Ether through regular transactions and throws an exception.

**Review by Flouse — DISMISSED** [review]

File: polyjuice-tests/src/test_cases/receive_ether.rs (hunk 2 above)
Comment (magicalne) [reply polyjuice-tests/src/test_cases/receive_ether.rs:22]: done

**Review by Flouse — APPROVED** [review]

**Comment by jjyr** [comment]
Can we add a test of transfer ethers more than the account had without input.

**Comment by magicalne** [comment]
> Can we add a test of transfer ethers more than the account had without input.

See https://github.com/nervosnetwork/godwoken-polyjuice/pull/155/commits/44fde5347d9dfea1224e845b05fc69de01685347
Not sure about this test case. Such over-transferring ethers have nothing to do with fallback() or receive(), maybe?

**Review by jjyr — APPROVED** [review]

**Review by Flouse — APPROVED** [review]

**Pull request merged**